mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -0,0 +1,571 @@
+
+      switch (inMemoryDataModel) {
+        case PIG:
+        case HIVE:
+          //TODO implement value readers for Pig and Hive
+          throw new UnsupportedOperationException();
+        case DEFAULT:
+          avroReadBuilder.createReaderFunc(DataReader::create);
+      }
+      return avroReadBuilder.build();
+    }
+
+    private CloseableIterable<T> newParquetIterable(InputFile inputFile, FileScanTask task, Schema readSchema) {
+      Parquet.ReadBuilder parquetReadBuilder = Parquet.read(inputFile)
+                                                      .project(readSchema)
+                                                      .filter(task.residual())
+                                                      .caseSensitive(caseSensitive)
+                                                      .split(task.start(), task.length());
+      if (reuseContainers) {
+        parquetReadBuilder.reuseContainers();
+      }
+
+      switch (inMemoryDataModel) {
+        case PIG:
+        case HIVE:
+          //TODO implement value readers for Pig and Hive
+          throw new UnsupportedOperationException();
+        case DEFAULT:
+          parquetReadBuilder.createReaderFunc(
+              fileSchema -> GenericParquetReaders.buildReader(readSchema, fileSchema));
+      }
+      return parquetReadBuilder.build();
+    }
+
+    private CloseableIterable<T> newOrcIterable(InputFile inputFile, FileScanTask task, Schema readSchema) {
+      ORC.ReadBuilder orcReadBuilder = ORC.read(inputFile)
+                                          .project(readSchema)
+                                          .caseSensitive(caseSensitive)
+                                          .split(task.start(), task.length());
+      // ORC does not support reuse containers yet
+      switch (inMemoryDataModel) {
+        case PIG:
+        case HIVE:
+          //TODO: implement value readers for Pig and Hive
+          throw new UnsupportedOperationException("In memory representation not yet supported for Pig and Hive");
+        case DEFAULT:
+          //TODO: We do not have support for Iceberg generics for ORC
+          throw new UnsupportedOperationException();
+      }
+
+      return orcReadBuilder.build();
+    }
+  }
+
+  private static Table findTable(Configuration conf) {
+    String path = conf.get(TABLE_PATH);
+    Preconditions.checkArgument(path != null, "Table path should not be null");
+    String catalogFuncClass = conf.get(CATALOG);
+    if (catalogFuncClass != null) {
+      Function<Configuration, Catalog> catalogFunc
+          = (Function<Configuration, Catalog>)
+          DynConstructors.builder(Function.class)
+                         .impl(catalogFuncClass)
+                         .build()
+                         .newInstance();
+      Catalog catalog = catalogFunc.apply(conf);
+      TableIdentifier tableIdentifier = TableIdentifier.parse(path);
+      return catalog.loadTable(tableIdentifier);
+    } else if (path.contains("/")) {
```

Comment:
I think this should be the first branch of the `if` because it is the easiest and is less likely to fail.